spec/services/cancel_rdv_by_agent_service.rb

```diff
@@ -0,0 +1,31 @@
+describe CancelRdvByAgentService, type: :service do
+  describe ".perform" do
+    let(:now) { Time.now }
+    let(:rdv) { create(:rdv) }
+
+    before do
+      travel_to(now)
+      freeze_time
+    end
+
+    subject { CancelRdvByAgentService.new(rdv).perform }
+
+    it "changes status to excused" do
+      expect { subject }.to change(rdv, :status).from("unknown").to("excused")
+    end
+
+    it "sets cancelled_at to now" do
+      expect { subject }.to change(rdv, :cancelled_at).from(nil).to(Time.now)
```

Comment:
if this is the only reason you're freezing time, I think it can be avoided by using the `be_within` matcher ie.

```
it "sets cancelled_at to now" do 
  expect(rdv.cancelled_at).to be_nil
  subject
  expect(rdv.cancelled_at).not_to be_nil
  expect(rdv.cancelled_at).to be_within(5.seconds).of(Time.now)
```

this matcher is useful for all float comparisons and such

if you can avoid freezing time it's always best because it can cause unexpected problems

Reply:
Ok, I used it in my first tests, but other devs preferred the freeze method.

Reply:
hahaha I'm SORRY in that case feel free to use whichever you prefer